secboot/secboot_tpm.go

```diff
@@ -313,7 +313,16 @@ type FDERevealKeyRequest struct {
 
 // fdeRevealKeyRuntimeMax is the maximum runtime a fde-reveal-key can execute
 // XXX: what is a reasonable default here?
-var fdeRevealKeyRuntimeMax = "2m"
+var fdeRevealKeyRuntimeMax = 2 * time.Minute
+
+// 50 ms means we check at a frequency 20 Hz, fast enough to not hold
+// up boot, but not too fast that we are hogging the CPU from the
+// thing we are waiting to finish running
+var fdeRevealKeyPollWait = 50 * time.Millisecond
+
+// fdeRevealKeyPollWaitParanoiaFactor controls much longer we wait
+// then fdeRevealKeyRuntimeMax before stopping to poll for results
+var fdeRevealKeyPollWaitParanoiaFactor = 2
```

Comment:
this factor seems a bit high to me, but maybe I'm under/overestimating slow devices?

Reply:
imho, this seems reasonable, if we consider the case where a process is hung and will never finish successfully (and systemd-run isn't able to kill it forcibly for some reason), this factor means we will take at max 4 minutes before failing and trickling that error back up to a screen, which seems reasonable as a user that if things are really bad you wait 5 minutes and then see some nasty error messages about what failed

on the flip side, where we have systemd-run misbehaving to the point where it is trying to kill it (but fails due to bugs) but the fde-reveal-key is progressing to the point where it will eventually finish, if we think that it could take up to 2 minutes to operate normally, then taking an additional 2 minutes if there is something really wrong, such as perhaps failing I/O or bad hw, or some really resource intensive thing, I think giving the device an extra 1-2 minutes is resaonable. I wouldn't give it much more time because this hook isn't supposed to be super resource intensive, but I have seen on occasion really slow devices take much longer than they should to do simple things because of silly things like the lack of vector assembly instructions meaning math takes a lot longer (which is maybe a bad example since that's a compile time thing, but hopefully you get the point that sometimes slow devices get really slow for silly reasons). I wouldn't give it much more than 1-2 minutes though because again this shouldn't be super time consuming to run this hook, and so if it's taking much more than 1-2 minutes extra it is probably just stuck and will never proceed.

all this is to say I think the factor of 2 is fine for specifically the case where the max runtime is 2 minutes, but if the max runtime was a different absolute value I don't think I would agree with the factor of 2, so perhaps what could be improved is that instead of multiplying the runtime by a factor we could add to the max runtime a set amount of "safety time" to let it do it's best. But that seems fine to do as a followup